hnn_core/network.py

```diff
@@ -509,8 +554,134 @@ def plot_input(self, ax=None, show=True):
+        aggregate network spiking activity according to cell type.
+    write(fname)
+        Write spiking activity to a collection of spike trial files.
+    '''
+
+    def __init__(self, times=None, gids=None, types=None):
+        if times is None:
+            times = []
+        if gids is None:
+            gids = []
+        if types is None:
+            types = []
+
+        # Validate arguments
+        arg_names = ['times', 'gids', 'types']
+        for arg_idx, arg in enumerate([times, gids, types]):
+            # Validate 1st-order list
+            if not isinstance(arg, list):
+                raise TypeError('%s should be a list of lists'
+                                % (arg_names[arg_idx],))
+            # If arg is not an empty list, validate 2st-order list
+            for trial_list in arg:
+                if not isinstance(trial_list, list):
+                    raise TypeError('%s should be a list of lists'
+                                    % (arg_names[arg_idx],))
+            # Set the length of 'times' as a references and validate
+            # uniform length
+            if arg == times:
+                n_trials = len(times)
+            if len(arg) != n_trials:
+                raise ValueError('times, gids, and types should be lists of '
+                                 'the same length')
+        self._times = times
+        self._gids = gids
+        self._types = types
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
+        n_trials = len(self._times)
+        return '<%s | %d simulation trials>' % (class_name, n_trials)
+
+    def __eq__(self, other):
+        if not isinstance(other, Spikes):
+            return NotImplemented
+        # Round each time element
+        times_self = [[round(time, 3) for time in trial]
+                      for trial in self._times]
+        times_other = [[round(time, 3) for time in trial]
+                       for trial in other._times]
+        return (times_self == times_other and
+                self._gids == other._gids and
+                self._types == other._types)
+
+    @property
+    def times(self):
+        return self._times
+
+    @property
+    def gids(self):
+        return self._gids
+
+    @property
+    def types(self):
+        return self._types
+
+    def update_types(self, gid_dict):
+        """Update spike types in the current instance of Spikes.
+
+        Parameters
+        ----------
+        gid_dict : dict of lists or range objects
+            Dictionary with keys 'evprox1', 'evdist1' etc.
+            containing the range of Cell or input IDs of different
+            cell or input types.
+        """
+
+        spiketypes = []
+        for trial_idx in range(len(self._times)):
+            spiketypes_trial = np.empty_like(self._times[trial_idx],
+                                             dtype='<U36')
+            for gidtype, gids in gid_dict.items():
```

Comment:
```suggestion
            for gid_type, gids in gid_dict.items():
```

I think it's better to put underscore so that it's clear where one word of the variable name ends and the second starts.